subprojects/launcher/src/main/java/org/gradle/launcher/daemon/server/health/HealthExpirationStrategy.java

```diff
@@ -85,12 +84,11 @@ public HealthExpirationStrategy(DaemonHealthStats stats, GarbageCollectorMonitor
         this.stats = stats;
         this.strategy = strategy;
         this.logger = logger;
-        this.enabled = Boolean.parseBoolean(System.getProperty(ENABLE_PERFORMANCE_MONITORING, "true"));
     }
 
     @Override
     public DaemonExpirationResult checkExpiration() {
-        if (!enabled) {
+        if (!Boolean.parseBoolean(System.getProperty(ENABLE_PERFORMANCE_MONITORING, "true"))) {
```

Comment:
💭 You should also add the note from the PR on why this can't be done during class initialization to this method, or the class javadoc or something.  

It would be really easy for someone working on this class later to think "Why do this work each time the method is called, I'll make a quick optimization to store the value in the constructor."  The test would fail, but they'd still lose any time trying to understand why.